lib/arclight/repository.rb

```diff
@@ -59,11 +59,18 @@ def self.all
       from_yaml(ENV['REPOSITORY_FILE'] || 'config/repositories.yml').values
     end
 
-    # Mimics ActiveRecord `find_by` behavior
+    # Mimics ActiveRecord dynamic `find_by` behavior for the slug or name
     #
-    # @param [String] `slug`
-    def self.find_by(slug)
-      all.select { |repo| repo.slug == slug }.first
+    # @param [String] `slug` or `name`
+    # @return [Repository]
+    def self.find_by(slug: nil, name: nil)
+      if slug
+        all.select { |repo| repo.slug == slug }.first
```

Comment:
Can we use `find` here instead? (and then we can drop `first` at the end).  I'm surprised rubocop doesn't complain about this.